OpenRA.Mods.Cnc/Traits/AttackPopupTurreted.cs

```diff
@@ -39,27 +39,33 @@ class AttackPopupTurretedInfo : AttackTurretedInfo, Requires<BuildingInfo>, Requ
 		public override object Create(ActorInitializer init) { return new AttackPopupTurreted(init, this); }
 	}
 
-	class AttackPopupTurreted : AttackTurreted, INotifyBuildComplete, INotifyIdle, IDamageModifier
+	class AttackPopupTurreted : AttackTurreted, INotifyBuildComplete, INotifyIdle, IDamageModifier, INotifyCreated
 	{
 		enum PopupState { Open, Rotating, Transitioning, Closed }
 
 		readonly AttackPopupTurretedInfo info;
 		readonly WithSpriteBody wsb;
-		readonly Turreted turret;
 
 		int idleTicks = 0;
 		PopupState state = PopupState.Open;
 		bool skippedMakeAnimation;
+		Turreted turret;
 
 		public AttackPopupTurreted(ActorInitializer init, AttackPopupTurretedInfo info)
 			: base(init.Self, info)
 		{
 			this.info = info;
-			turret = turrets.FirstOrDefault();
 			wsb = init.Self.Trait<WithSpriteBody>();
 			skippedMakeAnimation = init.Contains<SkipMakeAnimsInit>();
 		}
 
+		void INotifyCreated.Created(Actor self)
```

Comment:
Another way to fix the hack would be to trigger NotifyComplete in the frame-end task.